.github/workflows/build.yml

```diff
@@ -1,301 +1,293 @@
-        mode: ['http']
-    steps:
-      - uses: actions/setup-node@v3
-        with:
-          node-version: ${{ matrix.node }}
-      - uses: actions/download-artifact@v4
-        with:
-          name: web3-${{ matrix.node }}.js.tar.gz
-          path: /tmp
-      - run: tar -xf /tmp/web3-${{ matrix.node }}.js.tar.gz -C ./
-      - run: yarn test:e2e:${{ matrix.backend }}:${{ matrix.mode }}
-        shell: bash
-
-  e2e-ipc:
-    name: Integration # (geth with HTTP, IPC & WS)
-    needs: build
-    runs-on: ubuntu-latest
-    env:
-      INFURA_GOERLI_HTTP: ${{ secrets.INFURA_GOERLI_HTTP }}
-      INFURA_GOERLI_WS: ${{ secrets.INFURA_GOERLI_WS }}
-    strategy:
-      fail-fast: false
-      matrix:
-        node: [ 18 ]
-        backend: [ 'geth' ]
-        mode: [ 'ipc', 'ws', 'http' ]
-    steps:
-      -   uses: actions/setup-node@v3
-          with:
-            node-version: ${{ matrix.node }}
-      -   uses: actions/download-artifact@v4
-          with:
-            name: web3-${{ matrix.node }}.js.tar.gz
-            path: /tmp
-      -   run: tar -xf /tmp/web3-${{ matrix.node }}.js.tar.gz -C ./
-      -   run: yarn test:e2e:${{ matrix.backend }}:${{ matrix.mode }}
-          shell: bash
+                path: ./
+                key: web3-${{ matrix.node }}-${{github.event.pull_request.base.sha}}
 
-  e2e:
-    name: End-to-End ganache:ws
-    needs: build
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        node: [18]
-        browser: ['electron', 'chrome', 'firefox']
-    steps:
-      - uses: actions/setup-node@v3
-        with:
-            node-version: ${{ matrix.node }}
-      - uses: browser-actions/setup-firefox@latest
-        if: matrix.browser == 'firefox'
-      - uses: actions/download-artifact@v4
-        with:
-          name: web3-${{ matrix.node }}.js.tar.gz
-          path: /tmp
-      - run: tar -xf /tmp/web3-${{ matrix.node }}.js.tar.gz -C ./
-      - run: npm install --no-package-lock --no-save --force cypress
-      - name: Cypress run
-        uses: cypress-io/github-action@v4
-        with:
-            install: false
-            command: yarn test:e2e:ganache:ws:${{ matrix.browser }}
-            cache-key: node-v${{ matrix.node }}-on-${{ matrix.browser }}-hash-${{ hashFiles('yarn.lock') }}
+    build-esm:
+        name: Build ESM
+        needs: build
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                node: [ 18 ]
+        steps:
+            -   uses: actions/setup-node@v4
+                with:
+                    node-version: ${{ matrix.node }}
+            -   uses: actions/cache/restore@v3
+                with:
+                   path: ./
+                   key: web3-${{ matrix.node }}-${{github.event.pull_request.base.sha}}
+            -   run: yarn build:esm
+    build-types:
+        name: Build Types
+        needs: build
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
```

Comment:
there is only one node version so no need of matrix , same here and other places in this PR